main/src/ca/uwaterloo/flix/language/phase/jvm/GenUnitClass.scala

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2021 Jonathan Lindegaard Starup
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ca.uwaterloo.flix.language.phase.jvm
+
+import ca.uwaterloo.flix.api.Flix
+import ca.uwaterloo.flix.language.ast.FinalAst.Root
+import org.objectweb.asm.ClassWriter
+import org.objectweb.asm.Opcodes._
+
+object GenUnitClass {
+
+  val instanceMethodName: String = "getInstance"
```

Comment:
I think - in our spirit of simplifying things - we should simply have a static field called `INSTANCE` (following Java convention for enums) and not have any getter method at all. That is:

```
final class Unit {
  public static final Unit INSTANCE = new Unit();
  private Unit () {}
}
```

Reply:
> I think - in our spirit of simplifying things - we should simply have a static field called `INSTANCE` (following Java convention for enums) and not have any getter method at all. That is:
> 
> ...

Fixed now. Ready to merge